File: binding/include/libqgroupbox.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qgroupbox.html
pub const qgroupbox = struct {

    /// New constructs a new QGroupBox object.
    pub fn New(parent: ?*anyopaque) QtC.QGroupBox {
        return qtc.QGroupBox_new(@ptrCast(parent));
    }


    /// New2 constructs a new QGroupBox object.
    pub fn New2() QtC.QGroupBox {
        return qtc.QGroupBox_new2();
    }


    /// New3 constructs a new QGroupBox object.
    pub fn New3(title: []const u8) QtC.QGroupBox {
        const title_str = qtc.libqt_string{
    .len = title.len,
    .data = title.ptr,
};
return qtc.QGroupBox_new3(title_str);
    }


    /// New4 constructs a new QGroupBox object.
    pub fn New4(title: []const u8, parent: ?*anyopaque) QtC.QGroupBox {
        const title_str = qtc.libqt_string{
    .len = title.len,
    .data = title.ptr,
};
return qtc.QGroupBox_new4(title_str, @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGroupBox_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgroupbox.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Title(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGroupBox_Title(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgroupbox.Title: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetTitle(self: ?*anyopaque, title: []const u8) void {
        const title_str = qtc.libqt_string{
    .len = title.len,
    .data = title.ptr,
};
qtc.QGroupBox_SetTitle(@ptrCast(self), title_str);
    }

    pub fn Alignment(self: ?*anyopaque, ) i32 {
        return qtc.QGroupBox_Alignment(@ptrCast(self));
    }

    pub fn SetAlignment(self: ?*anyopaque, alignment: i32) void {
        qtc.QGroupBox_SetAlignment(@ptrCast(self), alignment);
    }

    pub fn MinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QGroupBox_MinimumSizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnMinimumSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QGroupBox_OnMinimumSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QGroupBox_QBaseMinimumSizeHint(@ptrCast(self));
    }

    pub fn IsFlat(self: ?*anyopaque, ) bool {
        return qtc.QGroupBox_IsFlat(@ptrCast(self));
    }

    pub fn SetFlat(self: ?*anyopaque, flat: bool) void {
        qtc.QGroupBox_SetFlat(@ptrCast(self), flat);
    }

    pub fn IsCheckable(self: ?*anyopaque, ) bool {
        return qtc.QGroupBox_IsCheckable(@ptrCast(self));
    }

    pub fn SetCheckable(self: ?*anyopaque, checkable: bool) void {
        qtc.QGroupBox_SetCheckable(@ptrCast(self), checkable);
    }

    pub fn IsChecked(self: ?*anyopaque, ) bool {
        return qtc.QGroupBox_IsChecked(@ptrCast(self));
    }

    pub fn SetChecked(self: ?*anyopaque, checked: bool) void {
        qtc.QGroupBox_SetChecked(@ptrCast(self), checked);
    }

    pub fn Clicked(self: ?*anyopaque, ) void {
        qtc.QGroupBox_Clicked(@ptrCast(self));
    }

    pub fn OnClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGroupBox_Connect_Clicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Toggled(self: ?*anyopaque, param1: bool) void {
        qtc.QGroupBox_Toggled(@ptrCast(self), param1);
    }

    pub fn OnToggled(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QGroupBox_Connect_Toggled(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Event(self: ?*anyopaque, event: ?*anyopaque) bool {
        return qtc.QGroupBox_Event(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QGroupBox_OnEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseEvent(self: ?*anyopaque, event: ?*anyopaque) bool {
        return qtc.QGroupBox_QBaseEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn ChildEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_ChildEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnChildEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGroupBox_OnChildEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseChildEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_QBaseChildEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn ResizeEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_ResizeEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnResizeEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGroupBox_OnResizeEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseResizeEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_QBaseResizeEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn PaintEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_PaintEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnPaintEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGroupBox_OnPaintEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBasePaintEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_QBasePaintEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn FocusInEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_FocusInEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnFocusInEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGroupBox_OnFocusInEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseFocusInEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_QBaseFocusInEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn ChangeEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_ChangeEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnChangeEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGroupBox_OnChangeEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseChangeEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_QBaseChangeEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn MousePressEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_MousePressEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnMousePressEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGroupBox_OnMousePressEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMousePressEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_QBaseMousePressEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn MouseMoveEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_MouseMoveEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnMouseMoveEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGroupBox_OnMouseMoveEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMouseMoveEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_QBaseMouseMoveEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn MouseReleaseEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_MouseReleaseEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnMouseReleaseEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGroupBox_OnMouseReleaseEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMouseReleaseEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QGroupBox_QBaseMouseReleaseEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn InitStyleOption(self: ?*anyopaque, option: ?*anyopaque) void {
        qtc.QGroupBox_InitStyleOption(@ptrCast(self), @ptrCast(option));
    }

    /// Allows for overriding the related default method
    pub fn OnInitStyleOption(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGroupBox_OnInitStyleOption(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseInitStyleOption(self: ?*anyopaque, option: ?*anyopaque) void {
        qtc.QGroupBox_QBaseInitStyleOption(@ptrCast(self), @ptrCast(option));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGroupBox_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgroupbox.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGroupBox_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgroupbox.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Clicked1(self: ?*anyopaque, checked: bool) void {
        qtc.QGroupBox_Clicked1(@ptrCast(self), checked);
    }

    pub fn OnClicked1(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QGroupBox_Connect_Clicked1(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QGroupBox_Delete(@ptrCast(self));
    }
};

File: binding/src/libqgroupbox.cpp
#include <QChildEvent>
#include <QEvent>
#include <QFocusEvent>
#include <QGroupBox>
#include <QMouseEvent>
#include <QPaintEvent>
#include <QResizeEvent>
#include <QSize>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QStyleOptionGroupBox>
#include <QWidget>
#include <qgroupbox.h>
#include "libqgroupbox.h"
#include "libqgroupbox.hxx"

QGroupBox* QGroupBox_new(QWidget* parent) {
	 return new VirtualQGroupBox(parent);
}

QGroupBox* QGroupBox_new2() {
	 return new VirtualQGroupBox();
}

QGroupBox* QGroupBox_new3(const libqt_string title) {
	 return new VirtualQGroupBox(QString::fromUtf8(title.data, title.len));
}

QGroupBox* QGroupBox_new4(const libqt_string title, QWidget* parent) {
	 return new VirtualQGroupBox(QString::fromUtf8(title.data, title.len), parent);
}

libqt_string QGroupBox_Tr(const char* s) {
	QString _ret = QGroupBox::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QGroupBox_Title(const QGroupBox* self) {
	QString _ret = self->title();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QGroupBox_SetTitle(QGroupBox* self, const libqt_string title) {
	self->setTitle(QString::fromUtf8(title.data, title.len));
}

int QGroupBox_Alignment(const QGroupBox* self) {
	return self->alignment();
}

void QGroupBox_SetAlignment(QGroupBox* self, int alignment) {
	self->setAlignment(alignment);
}

QSize* QGroupBox_MinimumSizeHint(const QGroupBox* self) {
	return new QSize(self->minimumSizeHint());
}

bool QGroupBox_IsFlat(const QGroupBox* self) {
	return self->isFlat();
}

void QGroupBox_SetFlat(QGroupBox* self, bool flat) {
	self->setFlat(flat);
}

bool QGroupBox_IsCheckable(const QGroupBox* self) {
	return self->isCheckable();
}

void QGroupBox_SetCheckable(QGroupBox* self, bool checkable) {
	self->setCheckable(checkable);
}

bool QGroupBox_IsChecked(const QGroupBox* self) {
	return self->isChecked();
}

void QGroupBox_SetChecked(QGroupBox* self, bool checked) {
	self->setChecked(checked);
}

void QGroupBox_Clicked(QGroupBox* self) {
	self->clicked();
}

void QGroupBox_Connect_Clicked(QGroupBox* self, intptr_t slot) {
    void (*slotFunc)(QGroupBox*) = reinterpret_cast<void (*)(QGroupBox*)>(slot);
    QGroupBox::connect(self, &QGroupBox::clicked, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QGroupBox_Toggled(QGroupBox* self, bool param1) {
	self->toggled(param1);
}

void QGroupBox_Connect_Toggled(QGroupBox* self, intptr_t slot) {
    void (*slotFunc)(QGroupBox*, bool) = reinterpret_cast<void (*)(QGroupBox*, bool)>(slot);
    QGroupBox::connect(self, &QGroupBox::toggled, [self, slotFunc](bool param1) {
	slotFunc(self, param1);
    });
}

libqt_string QGroupBox_Tr2(const char* s, const char* c) {
	QString _ret = QGroupBox::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QGroupBox_Tr3(const char* s, const char* c, int n) {
	QString _ret = QGroupBox::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QGroupBox_Clicked1(QGroupBox* self, bool checked) {
	self->clicked(checked);
}

void QGroupBox_Connect_Clicked1(QGroupBox* self, intptr_t slot) {
    void (*slotFunc)(QGroupBox*, bool) = reinterpret_cast<void (*)(QGroupBox*, bool)>(slot);
    QGroupBox::connect(self, &QGroupBox::clicked, [self, slotFunc](bool checked) {
	slotFunc(self, checked);
    });
}

// Base class handler implementation
QSize* QGroupBox_QBaseMinimumSizeHint(const QGroupBox* self) {
	auto* vqgroupbox = dynamic_cast<const VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_MinimumSizeHint_IsBase(true);
	return new QSize(vqgroupbox->minimumSizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QGroupBox_OnMinimumSizeHint(const QGroupBox* self, intptr_t slot) {
	auto* vqgroupbox = dynamic_cast<const VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_MinimumSizeHint_Callback(reinterpret_cast<VirtualQGroupBox::QGroupBox_MinimumSizeHint_Callback>(slot));
}
}

// Derived class handler implementation
bool QGroupBox_Event(QGroupBox* self, QEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
	return vqgroupbox->event(event);
	} else {
	return self->QGroupBox::event(event);
}
}

// Base class handler implementation
bool QGroupBox_QBaseEvent(QGroupBox* self, QEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_Event_IsBase(true);
	return vqgroupbox->event(event);
}
}

// Auxiliary method to allow providing re-implementation
void QGroupBox_OnEvent(QGroupBox* self, intptr_t slot) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_Event_Callback(reinterpret_cast<VirtualQGroupBox::QGroupBox_Event_Callback>(slot));
}
}

// Derived class handler implementation
void QGroupBox_ChildEvent(QGroupBox* self, QChildEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
	vqgroupbox->childEvent(event);
	} else {
	self->QGroupBox::childEvent(event);
}
}

// Base class handler implementation
void QGroupBox_QBaseChildEvent(QGroupBox* self, QChildEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_ChildEvent_IsBase(true);
	vqgroupbox->childEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QGroupBox_OnChildEvent(QGroupBox* self, intptr_t slot) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_ChildEvent_Callback(reinterpret_cast<VirtualQGroupBox::QGroupBox_ChildEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QGroupBox_ResizeEvent(QGroupBox* self, QResizeEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
	vqgroupbox->resizeEvent(event);
	} else {
	self->QGroupBox::resizeEvent(event);
}
}

// Base class handler implementation
void QGroupBox_QBaseResizeEvent(QGroupBox* self, QResizeEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_ResizeEvent_IsBase(true);
	vqgroupbox->resizeEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QGroupBox_OnResizeEvent(QGroupBox* self, intptr_t slot) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_ResizeEvent_Callback(reinterpret_cast<VirtualQGroupBox::QGroupBox_ResizeEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QGroupBox_PaintEvent(QGroupBox* self, QPaintEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
	vqgroupbox->paintEvent(event);
	} else {
	self->QGroupBox::paintEvent(event);
}
}

// Base class handler implementation
void QGroupBox_QBasePaintEvent(QGroupBox* self, QPaintEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_PaintEvent_IsBase(true);
	vqgroupbox->paintEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QGroupBox_OnPaintEvent(QGroupBox* self, intptr_t slot) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_PaintEvent_Callback(reinterpret_cast<VirtualQGroupBox::QGroupBox_PaintEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QGroupBox_FocusInEvent(QGroupBox* self, QFocusEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
	vqgroupbox->focusInEvent(event);
	} else {
	self->QGroupBox::focusInEvent(event);
}
}

// Base class handler implementation
void QGroupBox_QBaseFocusInEvent(QGroupBox* self, QFocusEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_FocusInEvent_IsBase(true);
	vqgroupbox->focusInEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QGroupBox_OnFocusInEvent(QGroupBox* self, intptr_t slot) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_FocusInEvent_Callback(reinterpret_cast<VirtualQGroupBox::QGroupBox_FocusInEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QGroupBox_ChangeEvent(QGroupBox* self, QEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
	vqgroupbox->changeEvent(event);
	} else {
	self->QGroupBox::changeEvent(event);
}
}

// Base class handler implementation
void QGroupBox_QBaseChangeEvent(QGroupBox* self, QEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_ChangeEvent_IsBase(true);
	vqgroupbox->changeEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QGroupBox_OnChangeEvent(QGroupBox* self, intptr_t slot) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_ChangeEvent_Callback(reinterpret_cast<VirtualQGroupBox::QGroupBox_ChangeEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QGroupBox_MousePressEvent(QGroupBox* self, QMouseEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
	vqgroupbox->mousePressEvent(event);
	} else {
	self->QGroupBox::mousePressEvent(event);
}
}

// Base class handler implementation
void QGroupBox_QBaseMousePressEvent(QGroupBox* self, QMouseEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_MousePressEvent_IsBase(true);
	vqgroupbox->mousePressEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QGroupBox_OnMousePressEvent(QGroupBox* self, intptr_t slot) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_MousePressEvent_Callback(reinterpret_cast<VirtualQGroupBox::QGroupBox_MousePressEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QGroupBox_MouseMoveEvent(QGroupBox* self, QMouseEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
	vqgroupbox->mouseMoveEvent(event);
	} else {
	self->QGroupBox::mouseMoveEvent(event);
}
}

// Base class handler implementation
void QGroupBox_QBaseMouseMoveEvent(QGroupBox* self, QMouseEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_MouseMoveEvent_IsBase(true);
	vqgroupbox->mouseMoveEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QGroupBox_OnMouseMoveEvent(QGroupBox* self, intptr_t slot) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_MouseMoveEvent_Callback(reinterpret_cast<VirtualQGroupBox::QGroupBox_MouseMoveEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QGroupBox_MouseReleaseEvent(QGroupBox* self, QMouseEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
	vqgroupbox->mouseReleaseEvent(event);
	} else {
	self->QGroupBox::mouseReleaseEvent(event);
}
}

// Base class handler implementation
void QGroupBox_QBaseMouseReleaseEvent(QGroupBox* self, QMouseEvent* event) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_MouseReleaseEvent_IsBase(true);
	vqgroupbox->mouseReleaseEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QGroupBox_OnMouseReleaseEvent(QGroupBox* self, intptr_t slot) {
	auto* vqgroupbox = dynamic_cast<VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_MouseReleaseEvent_Callback(reinterpret_cast<VirtualQGroupBox::QGroupBox_MouseReleaseEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QGroupBox_InitStyleOption(const QGroupBox* self, QStyleOptionGroupBox* option) {
	auto* vqgroupbox = dynamic_cast<const VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
	vqgroupbox->initStyleOption(option);
	} else {
	self->QGroupBox::initStyleOption(option);
}
}

// Base class handler implementation
void QGroupBox_QBaseInitStyleOption(const QGroupBox* self, QStyleOptionGroupBox* option) {
	auto* vqgroupbox = dynamic_cast<const VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_InitStyleOption_IsBase(true);
	vqgroupbox->initStyleOption(option);
}
}

// Auxiliary method to allow providing re-implementation
void QGroupBox_OnInitStyleOption(const QGroupBox* self, intptr_t slot) {
	auto* vqgroupbox = dynamic_cast<const VirtualQGroupBox*>(self);
	if (vqgroupbox && vqgroupbox->isVirtualQGroupBox) {
vqgroupbox->setQGroupBox_InitStyleOption_Callback(reinterpret_cast<VirtualQGroupBox::QGroupBox_InitStyleOption_Callback>(slot));
}
}

void QGroupBox_Delete(QGroupBox* self) {
    delete self;
}

